api/v1alpha1/datadogagent_types.go

```diff
@@ -518,6 +519,9 @@ type SystemProbeSpec struct {
 	// DebugPort Specify the port to expose pprof and expvar for system-probe agent
 	DebugPort int32 `json:"debugPort,omitempty"`
 
+	// EnableNetworkMonitoring enables Network Performance Monitoring
+	EnableNetworkMonitoring *bool `json:"enableNetworkMonitoring,omitempty"`
```

Comment:
For feature the Idea, is to put them under: `spec.features`  and try to apply always the same pattern

here for example
```yaml
spec:
  features:
     networkMonitoring:
        enable: true
```

the idea is to have a section for each feature. like this if this feature require later to add additional option we can aggregate in the section.

about the option naming. I don't know what should be the best name. We are already using `networkMonitoring` but the product is known as  NPM (NetworkPerformanceMonitoring). So maybe it can be good to as the network team what makes more sense for them.

Reply:
as example you have this PR: https://github.com/DataDog/datadog-operator/pull/268

Reply:
i made some updates, i hope it covers the bases but let me know if i missed something. (i checked with networks and they requested use of `networkMonitoring`)